salt/runners/queue.py

```diff
@@ -302,4 +303,6 @@ def process_runner(quantity=1, queue=None, backend=None):
     queue_kwargs = __get_queue_opts(queue=queue, backend=backend)
     data = process_queue(quantity=quantity, **queue_kwargs)
     for job in data['items']:
+        if backend=='sqlite':
+            job = json.loads(job.replace("'", '"'))
```

Comment:
I do not like handling this here by checking the backend.

I would much rather have a setting that allowed backends to set a dunder, or to move this logic to the queue backend.

Reply:
But backend don't have info — it was `insert_runner` or plain `insert` while runner_queue.process_queue call.

I can pass flag from process_runner => process_queue => backend if you like this. But maybe we can find another solution? :-)

Reply:
This is the biggest reason that the pgjsonb database is the only one supported right now, because we can just put it in.

IMO, if you want to use a local database file as storage for this, we should write queue storage using the built in python `shelve` module, instead of trying to get it to work in sqlite.

https://docs.python.org/3.6/library/shelve.html?highlight=shelve#shelve


Beyond that, we should just pass an argument to all the queues that says this is for storing a runner job, and then do the logic over there if it is needed.  That way if this fails, we can also say make a point to say this queue hasn't been tested for the job queue input.

Reply:
Ok , 
In this case maybe we should point more clearly what sqlite is unsupported for insert_runner ...

Sorry and close this PR plz :-(

Reply:
I still want to implement this!

You just always pass a value to the queue to tell it to do this logic for inserting the runner.

Then we also need to make each queue accept that argument for the insert function.

Reply:
Ok, I'll try implement this at next days :-)

Reply:
Hi,
Flag "is_runner" in commit